internal/github/publish_github_gist.go

```diff
@@ -0,0 +1,202 @@
+package github
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"log/slog"
+
+	gh "github.com/google/go-github/v58/github"
+	"github.com/hashicorp/hcl/v2"
+	"github.com/zclconf/go-cty/cty"
+	"go.opentelemetry.io/otel/attribute"
+	nooptrace "go.opentelemetry.io/otel/trace/noop"
+
+	"github.com/blackstork-io/fabric/pkg/diagnostics"
+	"github.com/blackstork-io/fabric/plugin"
+	"github.com/blackstork-io/fabric/plugin/dataspec"
+	"github.com/blackstork-io/fabric/plugin/dataspec/constraint"
+	"github.com/blackstork-io/fabric/print"
+	"github.com/blackstork-io/fabric/print/htmlprint"
+	"github.com/blackstork-io/fabric/print/mdprint"
+)
+
+func makeGithubGistPublisher(loader ClientLoaderFn) *plugin.Publisher {
+	return &plugin.Publisher{
+		Doc:  "Publishes content to github gist",
+		Tags: []string{},
+		Config: dataspec.ObjectSpec{
+			&dataspec.AttrSpec{
+				Name:        "github_token",
+				Type:        cty.String,
+				Constraints: constraint.RequiredNonNull,
+				Secret:      true,
+			},
+		},
+		Args: dataspec.ObjectSpec{
+			&dataspec.AttrSpec{
+				Name: "description",
+				Type: cty.String,
+			},
+			&dataspec.AttrSpec{
+				Name: "filename",
+				Type: cty.String,
+			},
+			&dataspec.AttrSpec{
+				Name: "make_public",
+				Type: cty.Bool,
+			},
+			&dataspec.AttrSpec{
+				Name: "gist_id",
+				Type: cty.String,
+			},
+		},
+		AllowedFormats: []plugin.OutputFormat{plugin.OutputFormatMD, plugin.OutputFormatHTML},
+		PublishFunc:    publishGithubGist(loader),
+	}
+}
+
+func parseContent(data plugin.MapData) (document *plugin.ContentSection) {
+	documentMap, ok := data["document"]
+	if !ok {
+		return
+	}
+	contentMap, ok := documentMap.(plugin.MapData)["content"]
+	if !ok {
+		return
+	}
+	content, err := plugin.ParseContentData(contentMap.(plugin.MapData))
+	if err != nil {
+		return
+	}
+	document = content.(*plugin.ContentSection)
+	return
+}
+
+func parseName(data plugin.MapData) (name string) {
+	documentMap, ok := data["document"]
+	if !ok {
+		return
+	}
+	metaMap, ok := documentMap.(plugin.MapData)["meta"]
+	if !ok {
+		return
+	}
+	documentName := metaMap.(plugin.MapData)["name"]
+	docName, ok := documentName.(plugin.StringData)
+	if !ok {
+		return
+	}
+	name = string(docName)
+	return
+}
+
+func publishGithubGist(loader ClientLoaderFn) plugin.PublishFunc {
+	// TODO: confirm if to be passed from the caller
+	logger := slog.New(slog.NewTextHandler(io.Discard, nil))
```

Comment:
```suggestion
	logger := slog.Default()
```

Inside of the plugin you can just use default slog logger, it is set-up to be proxied to the `fabric` binary. Dependancy injection happens there, and since plugins are incapable of being run on their own - there's no need to be fancy here.

Tracing is something that we haven't really done inside the plugins (afaik). Perhaps @dobarx knows how tracing plays with plugins.